modules/s3db/cap.py

```
@@ -296,7 +296,15 @@ def model(self):
                      Field("sender",
                            label = T("Sender"),
                            default = self.generate_sender,
-                           # @todo: can not be empty in alerts (validator!)
+                           requires = [IS_NOT_EMPTY(error_message="Sender cannot be empty for Alerts"),
+                                       IS_MATCH("^(.(?! ))*$",
+                                                error_message="Field cannot contain space"),
```

An alternative would be to silently encode the unwanted characters, e.g. with their escape sequences, or replace them with something similar, e.g. space by underscore, comma by semi-colon or period, & by +, < by I don't know.  Using the escapes has the advantage that they are displayable.

If we don't decide to silently replace the chars, we might want to handle this in the front end too, to ask the user to replace the chars if they include them.  But front end changes would be better delayed to after the switch to a person record.

Let's see if we can get a quick OK on escaping those chars rather than returning an error.  Would you like to ask on the mailing list?